pkg/publisher/local/publisher.go

```diff
@@ -24,18 +25,30 @@ type Publisher struct {
 	server        *LocalPublisherServer
 }
 
-func NewLocalPublisher(ctx context.Context, directory string, host string, port int) *Publisher {
+func NewLocalPublisher(ctx context.Context, directory string, host string, port int) (*Publisher, error) {
 	p := &Publisher{
 		baseDirectory: directory,
-		host:          ResolveAddress(ctx, host),
-		port:          port,
-		urlPrefix:     fmt.Sprintf("http://%s:%d", host, port),
+		// TODO: this field is only written to, never read. It could be deleted.
+		host:      ResolveAddress(ctx, host),
+		port:      port,
+		urlPrefix: fmt.Sprintf("http://%s:%d", host, port),
 	}
 
+	if info, err := os.Stat(p.baseDirectory); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, fmt.Errorf("failed to create local pubisher with path %s: path could not be read: %w", directory, err)
+		}
+		log.Warn().Msgf("local publisher was configured with a directory that doesn't exits. attempting to create one at %s", directory)
+		if err := os.MkdirAll(p.baseDirectory, util.OS_USER_RWX); err != nil {
+			return nil, fmt.Errorf("failed to create directory for local publisher: %w", err)
+		}
+	} else if !info.IsDir() {
+		return nil, fmt.Errorf("failed to create local publisher with path: %s: path is not a directoy", directory)
+	}
 	p.server = NewLocalPublisherServer(ctx, p.baseDirectory, p.port)
 	go p.server.Run(ctx)
 
-	return p
+	return p, nil
 }
```

Comment:
There are instance in testing where the path of the directory doesn't exist, here we attempt to create it in the event it DNE.